File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/Log.java
package ru.iqchannels.sdk;

public class Log {
    private static boolean LOGGING = true;

    public static void configure(boolean state) {
        synchronized(Log.class) {
            LOGGING = state;
        }
    }

    public static void d(String tag, String message) {
        if (LOGGING) android.util.Log.d(tag, message);
    }

    public static void i(String tag, String message) {
        d(tag, message);
    }

    public static void e(String tag, String message) {
        if (LOGGING) android.util.Log.e(tag, message);
    }
}

File: iqchannels-sdk/src/main/java/ru/iqchannels/sdk/Log.kt
package ru.iqchannels.sdk

object Log {

	private var LOGGING = true

	fun configure(state: Boolean) {
		synchronized(Log::class.java) { LOGGING = state }
	}

	fun d(tag: String, message: String) {
		if (LOGGING) android.util.Log.d(tag, message)
	}

	fun i(tag: String, message: String) {
		d(tag, message)
	}

	@JvmStatic
	fun e(tag: String, message: String) {
		if (LOGGING) android.util.Log.e(tag, message)
	}
}
